### DISCUSSION TITLE
PnP "Vendor Ids" with [ISA16] Crystal CS4235 and [ISA16] Crystal CS4236B

### DISCUSSION THREAD

**Discussion by CryptusDE** [86Box/86Box#4280] [Q&A] [answered]
I am using 86Box build 5497 right now, great work, and want to thank all people developing this software.
However, I have a question concerning the above mentioned Crystal soundcards.

I understand that there are quite a lot of different PnP Vendor Ids with Crystal soundcards, not always corresponding to the chip.
But with 86Box, Crystal CS4235 has Vendor Id "CSC4225" (sic!) and Crystal CS4236B has Vendor Id "CSC4235" (sic!).

While this does not affect functionality and/or driver loading etc. in any way, the output of detection/initialization software (like pnpdump.exe or UNISOUND driver under DOS, only to mention two, or similar programs under Linux) is at least a little confusing.

At the moment my DOS software has to undergo a number of steps (in batch files), like:
Am I running under real hardware or under 86Box? (detection process...)
Oh, I am under 86Box, so I must know that in this case PnP Vendor Id "CSC4225" means CS4235 and Vendor Id "CSC4235" means CS4236.

Of course this works.

However, the question arises:
Is there any special reason for this strange Vendor Id labeling (i.e. it's a "feature"),
or is it only the result of some rapidly written code (i.e. just "annoying", to avoid the harsh term "bug")?

**Comment by OBattler** [comment, marked as answer]
These Vendor ID's came from Crystal's own datasheets, based on which the emulation was implemented.

**Comment by CryptusDE** [comment]
Ok, thanks for the clarification!
